clients/venice-producer/src/main/java/com/linkedin/venice/producer/online/ProducerTool.java

```diff
@@ -0,0 +1,226 @@
+package com.linkedin.venice.producer.online;
+
+import static com.linkedin.venice.CommonConfigKeys.SSL_FACTORY_CLASS_NAME;
+import static com.linkedin.venice.VeniceConstants.DEFAULT_SSL_FACTORY_CLASS_NAME;
+
+import com.linkedin.avroutil1.compatibility.AvroCompatibilityHelper;
+import com.linkedin.venice.client.schema.RouterBasedStoreSchemaFetcher;
+import com.linkedin.venice.client.store.AbstractAvroStoreClient;
+import com.linkedin.venice.client.store.AvroGenericStoreClient;
+import com.linkedin.venice.client.store.ClientConfig;
+import com.linkedin.venice.client.store.ClientFactory;
+import com.linkedin.venice.client.store.StatTrackingStoreClient;
+import com.linkedin.venice.exceptions.VeniceException;
+import com.linkedin.venice.schema.vson.VsonAvroSchemaAdapter;
+import com.linkedin.venice.security.SSLFactory;
+import com.linkedin.venice.utils.SslUtils;
+import com.linkedin.venice.utils.VeniceProperties;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.util.Properties;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericDatumReader;
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.HelpFormatter;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+import org.apache.commons.lang.StringUtils;
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.core.config.Configurator;
+
+
+public class ProducerTool {
+  private static final Option STORE_OPTION =
+      Option.builder().option("s").longOpt("store").hasArg().required().desc("Store name").build();
+  private static final Option KEY_OPTION =
+      Option.builder().option("k").longOpt("key").hasArg().required().desc("Key string").build();
```

Comment:
Should we term it as "string" when it can be one of many types as in `adaptDataToSchema`.

Reply:
Hmm, it's tricky. We can call it just key and mention in the description that it accepts json formatted data for complex types

Reply:
That sounds good.